src/main/java/org/opensearch/flowframework/workflow/SearchResponseProcessorStep.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.support.PlainActionFuture;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.flowframework.util.ParseUtils;
+import org.opensearch.search.pipeline.Processor;
+import org.opensearch.search.pipeline.Processor.PipelineContext;
+import org.opensearch.search.pipeline.Processor.PipelineSource;
+import org.opensearch.search.pipeline.SearchPipelineService;
+import org.opensearch.search.pipeline.SearchResponseProcessor;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+import static org.opensearch.flowframework.common.CommonValue.DESCRIPTION_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.PROCESSOR_CONFIG;
+import static org.opensearch.flowframework.common.CommonValue.SEARCH_REQUEST;
+import static org.opensearch.flowframework.common.CommonValue.SEARCH_RESPONSE;
+import static org.opensearch.flowframework.common.CommonValue.TAG;
+import static org.opensearch.flowframework.common.CommonValue.TYPE;
+
+/**
+ * Step to create a search response processor
+ */
+public class SearchResponseProcessorStep implements WorkflowStep {
```

Comment:
Can you name it as ReRankProcessorStep for better clarity?

Reply:
Its a general step to create any search response processor, the type of processor created depends on the processor config passed to it